### PR TITLE
Optimistic updates

### PR DESCRIPTION
**Pull request by salman-aziz-4425**

### CODE REVIEW

File: client/components/pack/AddPack.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const AddPack = () => {
    * @return {void}
    */
   const handleAddPack = () => {
-    dispatch(addPack({ name, owner_id: user?._id, is_public: isPublic }));
+    mutation.mutate({ name, owner_id: user?._id, is_public: isPublic })
```
Comment (andrew-bierman) [inline client/components/pack/AddPack.js:42]: we should be returning mutation.mutate in the hook so we can call addPack like you did for edit pack item

File: client/hooks/packs/useAddNewPack.js [hunk 2]
```diff
@@ -1,6 +1,61 @@
+import { getQueryKey } from '@trpc/react-query';
+import { queryClient } from '~/constants/queryClient';
```
Comment (andrew-bierman) [inline client/hooks/packs/useAddNewPack.js:2]: if we need query client we should be importing the trpc one for consistency (not your fault i know its confusing how we have multiple created in app at the moment)

File: client/components/pack/AddPack.js (hunk 1 above)
Comment (salman-aziz-4425) [reply client/components/pack/AddPack.js:42]: Ok i will push it again along with delete pack item

File: client/hooks/packs/useAddNewPack.js (hunk 2 above)
Comment (salman-aziz-4425) [reply client/hooks/packs/useAddNewPack.js:2]: I mistakenly added this i am only using use context majorly will remove it

**Pull request merged**